lib/hawkular/inventory/inventory_api_v4.rb

```diff
@@ -37,35 +36,47 @@ def self.create(hash)
     # Get single resource by id
     # @return Resource the resource
     def resource(id)
-      hash = http_get("/resources/#{URI.escape(id)}")
+      hash = http_get(url('/resources/%s', id))
       Resource.new(hash)
     end
 
     # Get resource by id with its complete subtree
     # @return Resource the resource
     def resource_tree(id)
-      hash = http_get("/resources/#{URI.escape(id)}/tree")
+      hash = http_get(url('/resources/%s/tree', id))
       Resource.new(hash)
     end
 
     # Get childrens of a resource
     # @return Children of a resource
     def children_resources(parent_id)
-      http_get("/resources/#{parent_id}/children")['results'].map { |r| Resource.new(r) }
+      http_get(url('/resources/%s/children', parent_id))['results'].map { |r| Resource.new(r) }
     end
 
     # List root resources
     # @return [Array<Resource>] List of resources
     def root_resources
       # FIXME: pagination => lazy-loaded list with ruby?
-      http_get('/resources?root=true')['results'].map { |r| Resource.new(r) }
+      resources root: "true"
+    end
+
+    # List resources
+    # @param [Hash] filter options to filter the resource list
+    # @option filter :root If truthy, only get root resources
+    # @option filter :feedId Filter by feed id
+    # @option filter :typeId Filter by type id
+    def resources(filter = {})
+      # FIXME: pagination => lazy-loaded list with ruby?
+      filter[:root] = !! filter[:root] if filter.key? :root
+      filter_query = '?' + filter.keys.join('=%s&') + '=%s' unless filter.empty?
+      http_get(url("/resources#{filter_query}", *filter.values))['results'].map { |r| Resource.new(r) }
     end
 
     # List resources for type
     # @return [Array<Resource>] List of resources
     def resources_for_type(type)
       # FIXME: pagination => lazy-loaded list with ruby?
-      http_get("/resources?typeId=#{URI.escape(type)}")['results'].map { |r| Resource.new(r) }
+      resources typeId: "type"
```

Comment:
`type`, not `"type"` (ie. not a string), I guess

Reply:
Whoops